File: 프로그래머스/2/42586. 기능개발/기능개발.py
from collections import deque

def solution(progresses, speeds):
    answer = []
    
    for_done = deque([]) # 진도 100% 되는 일수
    for i in range(len(progresses)):
        left = 100 - progresses[i]
        for_done.append(left // speeds[i] if left % speeds[i] == 0 else left // speeds[i] + 1)
    
    cnt = 1
    start = for_done.popleft() # 먼저 배포되어야 하는 작업의 완료 시점
    while for_done:
        nxt = for_done.popleft()
        if nxt <= start:
            cnt += 1
        else:
            answer.append(cnt)
            start = nxt
            cnt = 1
    answer.append(cnt)
    return answer

File: 프로그래머스/2/42586. 기능개발/기능개발.java
import java.util.*;

class Solution {
    public int[] solution(int[] progresses, int[] speeds) {
        List<Integer> answer = new ArrayList<>();
        
        Queue<Integer> daysNeeded = new LinkedList<>();
        for (int i = 0; i < progresses.length; i++) {
            int remaining = 100 - progresses[i];
            daysNeeded.add((int) Math.ceil((double) remaining / speeds[i]));
        }
        
        int count = 1;
        int current = daysNeeded.poll(); // 먼저 배포되어야 하는 작업의 완료 시점
        while(!daysNeeded.isEmpty()) {
            int next = daysNeeded.poll();
            if (next <= current) {
                count++;
            } else {
                answer.add(count);
                current = next;
                count = 1;
            }
        }
        answer.add(count);
        
        return answer.stream().mapToInt(i -> i).toArray();
    }
}